torchgeo/datamodules/deepglobelandcover.py

```diff
@@ -40,14 +40,23 @@ def __init__(
             **kwargs: Additional keyword arguments passed to
                 :class:`~torchgeo.datasets.DeepGlobeLandCover`.
         """
-        super().__init__(DeepGlobeLandCover, 1, num_workers, **kwargs)
+        super().__init__(
+            DeepGlobeLandCover, batch_size=batch_size, num_workers=num_workers, **kwargs
+        )
 
         self.patch_size = _to_tuple(patch_size)
         self.val_split_pct = val_split_pct
 
         self.aug = K.AugmentationSequential(
             K.Normalize(mean=self.mean, std=self.std),
-            _RandomNCrop(self.patch_size, batch_size),
+            _ExtractPatches(window_size=self.patch_size),
+            data_keys=None,
+            keepdim=True,
+            same_on_batch=True,
+        )
+        self.train_aug = K.AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
+            K.RandomCrop(self.patch_size, pad_if_needed=True),
```

Comment:
Do we need `pad_if_needed` or is that just because of our fake test data being small?

Reply:
Yes, or if the image size is not evenly divisible by the patch size it will pad appropriately in the bottom right of the patch.